mypy/checkexpr.py

```diff
@@ -2822,6 +2818,13 @@ def check_boolean_op(self, e: OpExpr, context: Context) -> Type:
             assert right_map is not None  # find_isinstance_check guarantees this
             return right_type
 
+        if e.op == 'and':
```

Comment:
Can safely move it down here since nothing uses `restricted_left_type` and `result_is_left` until this point.